### PR TITLE
Add tls secret option

### PR DESCRIPTION
**Pull request by soneillf5**
### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-ingress-operator/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/nginx-ingress-controller.md [hunk 1]
```diff
@@ -147,6 +147,7 @@ spec:
 | --- | --- | --- | --- |
 | `enable` | `boolean` | Enable Prometheus metrics. | Yes |
 | `port` | `int` | Sets the port where the Prometheus metrics are exposed. Default is 9113. Format is `1023 - 65535`. | No |
+| `secret` | `string` | Sets the namespace/name of a TLS Secret Resource to use to enable TLS for the Prometheus endpoint. | No |
```
Comment (vepatel) [inline docs/nginx-ingress-controller.md:150]: think need to add arg above line 76 too

File: config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml [hunk 2]
```diff
@@ -197,6 +197,13 @@ spec:
                     minimum: 1023
                     nullable: true
                     type: integer
+                  secret:
+                    description: A Secret with a TLS certificate and key for TLS termination
```
Comment (lucacome) [inline config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml:201]: This description (and the one in bundle) doesn't match the comment in `types`. I think you need to run `make bundle` again.

**Review by lucacome — DISMISSED** [review]

File: config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml (hunk 2 above)
Comment (soneillf5) [reply config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml:201]: Ah I didn't realise it was generated. I ran `make bundle` and got the following error:
make bundle
```
/Users/soneill/go/src/github.com/nginxinc/nginx-ingress-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
operator-sdk generate kustomize manifests -q
FATA[0000] failed to read config: error unmarshalling project configuration: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go struct field Config.layout of type string
make: *** [bundle] Error 1
```
I've update this PR anyway with he generated changes.

File: docs/nginx-ingress-controller.md (hunk 1 above)
Comment (soneillf5) [reply docs/nginx-ingress-controller.md:151]: I've updated the example if that's what you mean? I don't see anywhere else to add it

Comment (vepatel) [reply docs/nginx-ingress-controller.md:151]: yep that was it

**Review by vepatel — APPROVED** [review]

**Review by ciarams87 — APPROVED** [review]

File: config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml (hunk 2 above)
Comment (lucacome) [reply config/crd/bases/k8s.nginx.org_nginxingresscontrollers.yaml]: The files in bundle are auto-generated too.

**Pull request merged**